apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifimanager_main.c

```diff
@@ -111,283 +117,267 @@ void wifi_scan_ap_done(wifi_manager_cb_msg_s msg, void *arg)
-
-	ret = wifi_manager_set_mode(SOFTAP_MODE, &ap_config);
-
-	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
-
-	ret = wifi_manager_set_mode(STA_MODE, NULL);
-
-	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
-
+	TC_ASSERT_EQ("wifi_manager_init", wifi_manager_init(&wifi_callbacks), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_set_mode", wifi_manager_set_mode(SOFTAP_MODE, &g_softap_config), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_set_mode", wifi_manager_set_mode(STA_MODE, NULL), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager deinit", wifi_manager_deinit(), WIFI_MANAGER_SUCCESS);
 	TC_SUCCESS_RESULT();
 }
 
 static void utc_wifimanager_get_mode_n(void)
 {
-	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
-	wifi_manager_info_s *info = NULL;
-
-	ret = wifi_manager_get_info(info);
-
-	TC_ASSERT_EQ("wifi_manager_get_info", ret, WIFI_MANAGER_INVALID_ARGS);
+	TC_ASSERT_EQ("wifi_manager_init", wifi_manager_init(&wifi_callbacks), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_get_info", wifi_manager_get_info(NULL), WIFI_MANAGER_INVALID_ARGS);
+	TC_ASSERT_EQ("wifi_manager deinit", wifi_manager_deinit(), WIFI_MANAGER_SUCCESS);
 	TC_SUCCESS_RESULT();
 }
 
 static void utc_wifimanager_get_mode_p(void)
 {
-	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
 	wifi_manager_info_s info;
-	int ret_cmp = 0;
-	int i;
-
-	ret = wifi_manager_get_info(&info);
-
-	TC_ASSERT_EQ("wifi_manager_get_info", ret, WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_init", wifi_manager_init(&wifi_callbacks), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_get_info", wifi_manager_get_info(&info), WIFI_MANAGER_SUCCESS);
 	TC_ASSERT_EQ("wifi_manager_get_info", info.mode, STA_MODE);
-
-	wifi_manager_softap_config_s ap_config;
-	strncpy(ap_config.ssid, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_SSID,
-			strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_SSID) + 1);
-	ap_config.channel = CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_CHANNEL;
-	strncpy(ap_config.passphrase, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_PASSPHRASE,
-			strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_PASSPHRASE) + 1);
-	ret = wifi_manager_set_mode(SOFTAP_MODE, &ap_config);
-
-	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
-
-	ret = wifi_manager_get_info(&info);
-
-	TC_ASSERT_EQ("wifi_manager_get_info", ret, WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_set_mode", wifi_manager_set_mode(SOFTAP_MODE, &g_softap_config), WIFI_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("wifi_manager_get_info", wifi_manager_get_info(&info), WIFI_MANAGER_SUCCESS);
 	TC_ASSERT_EQ("wifi_manager_get_info", info.mode, SOFTAP_MODE);
-
-	ret_cmp = strncmp(info.ssid, ap_config.ssid, strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC_SOFTAP_SSID));
-	TC_ASSERT_EQ("wifi_manager_get_mode", ret_cmp, 0);
-
+	TC_ASSERT_EQ("wifi_manager_get_mode", strncmp(info.ssid, g_softap_config.ssid, strlen(g_softap_config.ssid)), 0);
```

Comment:
I think it is better to compare with `strlen() + 1` for null termination checking.

Reply:
Thanks I'll modify it